src/tests/tracing/runtimeeventsource/NativeRuntimeLevelKeywords.cs

```diff
@@ -0,0 +1,285 @@
+﻿//*
+#define GC_COLLECT
+//*/
+//*
+#define GC_FINALIZERS
+//*/
+//*
+#define GC_MEMORY_PRESSURE
+//*/
+/*
+#define LOCK_CONTENTION //BUG!? Possibly deadlocks Linux x64
+//*/
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Linq;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+#if LOCK_CONTENTION
+using System.Threading;
+using System.Threading.Tasks;
+#endif //LOCK_CONTENTION
+
+using System.Buffers; // Copied from https://github.com/dotnet/runtime/pull/86370 by @davmason:
+{
+    // Access ArrayPool.Shared.Rent() before the test to avoid the deadlock reported
+    // in https://github.com/dotnet/runtime/issues/86233. This is a real issue,
+    // but only seen if you have a short lived EventListener and create EventSources
+    // in your OnEventWritten callback so we don't expect customers to hit it.
+    byte[] localBuffer = ArrayPool<byte>.Shared.Rent(10);
+    Console.WriteLine($"buffer length={localBuffer.Length}");
+}
+
+const EventKeywords keywords = EventKeywords.None
+#if GC_COLLECT || GC_FINALIZERS || GC_MEMORY_PRESSURE
+    | (EventKeywords)0x1
+#endif
+#if LOCK_CONTENTION
+    | (EventKeywords)0x4000
+#endif //LOCK_CONTENTION
+    ;
+using Listener informational = Listener.StartNew("Microsoft-Windows-DotNETRuntime", EventLevel.Informational, keywords);
+using Listener verbose = Listener.StartNew("Microsoft-Windows-DotNETRuntime", EventLevel.Verbose, keywords);
+using Listener logAlways = Listener.StartNew("Microsoft-Windows-DotNETRuntime", EventLevel.LogAlways, keywords);
+
+#if GC_COLLECT
+for (int i = 0; i < ushort.MaxValue; i++)
+{
+    GC.KeepAlive(new());
+}
+GC.Collect();
+#endif //GC_COLLECT
+
+#if GC_FINALIZERS
+GC.WaitForPendingFinalizers();
+#endif //GC_FINALIZERS
+
+#if GC_MEMORY_PRESSURE
+GC.AddMemoryPressure(1L);
+GC.RemoveMemoryPressure(1L);
+#endif //GC_MEMORY_PRESSURE
+
+#if LOCK_CONTENTION
+for (int i = 1; i <= 0x40; i++)
+{
+    object lockObject = new();
+    Task task = new(() =>
+    {
+        lock (lockObject)
+        {
+            Thread.Sleep(0);
+        }
+    });
+    lock (lockObject)
+    {
+        task.Start();
+        while (Monitor.LockContentionCount < i)
+        {
+            Thread.Sleep(0);
+        }
+    }
+    task.Wait();
+}
+#endif //LOCK_CONTENTION
+
+informational.DumpEvents();
+verbose.DumpEvents();
+logAlways.DumpEvents();
+
+IEnumerable<Listener.Event> informationalEvents = new Listener.Event[]
+{
+#if GC_COLLECT
+
+#if false //BUG!? Shouldn’t events be consistent across all platforms?
+    new(1,   EventLevel.Informational, (EventKeywords)0x0000F00000000001, "GCStart_V2"),
```

Comment:
This could be simplified, instead of maintaining lists of events you could just pick an event or two and have a bool in the listener that keeps track of whether you saw those events. We don't need to verify that you see every event - we aren't testing the GC events, we are testing the event level filtering logic.

So you could take one informational event that always happens (e.g. GCStart_V2), and one verbose event that always happens (e.g. GCAllocationTick_V4) and make sure the right listeners see at least one of the events.

There is always some variability in the total number of events received, and changes to the runtime can make the test invalid over time. So keeping it as simple as possible and only testing the thing we intend to is a good strategy.